app/src/main/java/emu/skyline/EmulationActivity.kt

```diff
@@ -480,6 +499,24 @@ class EmulationActivity : AppCompatActivity(), SurfaceHolder.Callback, View.OnTo
         }
     }
 
+    /**
+    * Updating the layout depending on type and state of device
+    */
+    private fun updateCurrentLayout(newLayoutInfo: WindowLayoutInfo) {
+        if (!emulationSettings.supportFoldableScreen) return
+        binding.onScreenGameView.layoutParams.height = ViewGroup.LayoutParams.MATCH_PARENT
+        requestedOrientation = emulationSettings.orientation
+        val foldingFeature = newLayoutInfo.displayFeatures.find { it is FoldingFeature }
+        (foldingFeature as? FoldingFeature)?.let {
+            if (it.isSeparating) {
+                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
+                if (it.orientation == FoldingFeature.Orientation.HORIZONTAL)
+                    binding.onScreenGameView.layoutParams.height = it.bounds.top
+            }
+        }
```

Comment:
Would love to see some comments in here to explaining the **why** of what is being done. Unlikely anyone will remember months from now, let alone years in the future.